java/src/jmri/jmrix/AbstractMRTrafficController.java

```diff
@@ -33,15 +33,24 @@
  */
 abstract public class AbstractMRTrafficController {
 
+    private Thread shutdownHook = null; // retain shutdown hook for 
+                                        // possible removal.
+
     public AbstractMRTrafficController() {
         log.debug("Creating AbstractMRTrafficController instance");
         mCurrentMode = NORMALMODE;
         mCurrentState = IDLESTATE;
         allowUnexpectedReply = false;
         setInstance();
-        // should this be a ShutDownTask instead, or are we worried at this point
-        // that a ShutDownManager does not yet exist?
-        Runtime.getRuntime().addShutdownHook(new Thread(new CleanupHook(this)));
+
+        // We use a shutdown hook here to make sure the connection is left
+        // in a clean state prior to exiting.  This is required on systems
+        // which have a service mode to ensure we don't leave the system 
+        // in an unusable state (This code predates the ShutdownTask 
+        // mechanisim, but it remains a shutdown hook because shtudown hooks
+        // will always run after ShutdownTasks do).
```

Comment:
Note that this comment is incorrect, since the running of the ShutdownTasks may be triggered from a shutdown hook.